tests/test_submit.py

```diff
@@ -0,0 +1,114 @@
+    from pytest_mock import MockerFixture
+
+class CheckoutGitTestScenario(BaseModel):
+    debug_scenario_name: str
+    git_status_mocked_output: str = ""
+    internet_permissions: list[str] = []
+    git_push_result: tuple[int, str] | None = None # (exit_code, output_message)
+    expected_prompts: list[str] = []
+
+@pytest.fixture(autouse=True)
+def clear_modules():
+    import sys
+    if "src.submit" in sys.modules:
+        del sys.modules["src.submit"]
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "scenario",
+    [
+        CheckoutGitTestScenario(
+            debug_scenario_name="no_changes_with_internet",
+            internet_permissions=["internet"],
+            git_push_result=(0, "Everything up-to-date"),
+            expected_prompts=[],
+        ),
+        CheckoutGitTestScenario(
+            debug_scenario_name="no_changes_without_internet",
+            internet_permissions=[],
+            git_push_result=None,
+            expected_prompts=[
+                "Since this task is running on a container with no internet access, please clone the repository to your local machine and push your changes from there to github, and only confirm once this is done."
+            ],
+        ),
+        CheckoutGitTestScenario(
+            debug_scenario_name="uncommitted_changes_confirmed",
+            git_status_mocked_output="M modified_file.txt",
+            internet_permissions=[],
+            git_push_result=None,
+            expected_prompts=[
+                "Are you sure you want to continue?",
+                "Since this task is running on a container with no internet access, please clone the repository to your local machine and push your changes from there to github, and only confirm once this is done.",
+            ],
+        ),
+        CheckoutGitTestScenario(
+            debug_scenario_name="push_failure_confirmed",
+            internet_permissions=["internet"],
+            git_push_result=(1, "Failed to push: remote error"),
+            expected_prompts=["Are you sure you want to continue?"],
+        ),
+    ]
+)
+async def test_check_git_repo(
+    mocker: MockerFixture,
+    capsys: pytest.CaptureFixture,
+    scenario: CheckoutGitTestScenario,
+):
+    # Mock HOOKS.getTask (before importing submit)
+    mock_hooks = mocker.Mock()
+    mock_task = mocker.Mock(permissions=scenario.internet_permissions)
+    mock_hooks.getTask = mocker.AsyncMock(return_value=mock_task)
+    mocker.patch("src.settings.HOOKS", mock_hooks)
+
+    from src.submit import _check_git_repo
+
+    # Mock git status (subprocess.create_subprocess_exec)
+    async def mock_subprocess_exec(*args, **kwargs):
+        process_mock = mocker.AsyncMock()
+        process_mock.communicate.return_value = (scenario.git_status_mocked_output.encode(), None)
+        process_mock.wait.return_value = 0
+        return process_mock
+    mocker.patch("asyncio.subprocess.create_subprocess_exec", side_effect=mock_subprocess_exec)
```

Comment:
We use https://pypi.org/project/pytest-subprocess/ in our other repos for stuff like this

Reply:
(seems like I missed this comment and some others since the lines were changed, my bad)

Reply:
I changed it, but note this also means I moved us away from using `asyncio.subprocess` to `subprocess`, see [here](https://github.com/poking-agents/headless-human/pull/29/commits/675b92d5211eda241f917e44e4d18929da8643df#diff-1d6580cb8bf6e240aa0206238a5d2676f36f89d6a161c5cc82794d777b7379bd)

Reply:
Oh, it does have asyncio support, hold on

Reply:
Went back to using asyncio.

The problem was we used `asyncio.subprocess.create_subprocess_exec` instead of `asyncio.create_subprocess_exec`